### PR TITLE
[pytorch][build][E3] Update PyTorch 1.11.0 cuda 11.3 Dockerfiles

### PR DESCRIPTION
**Pull request by ydaiming**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

Introduce CUDA 11.3 version of PyTorch 1.11.0 E3 DLCs, preparing for the CUDA 11.3 version of PyTorch 1.11.0 SageMaker DLCs, as SageMaker doesn't support CUDA 11.5 yet. Based on initial investigations, CUDA 11.3 performs as good as CUDA 11.5 when the same CUDA driver is used.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu [hunk 1]
```diff
@@ -124,10 +124,14 @@ RUN mkdir /tmp/efa \
...
  && cd aws-efa-installer \
+ && apt-get update \
+ && apt-get upgrade -y \
  && ./efa_installer.sh -y --skip-kmod -g \
  && rm -rf $OPEN_MPI_PATH \
  && rm -rf /tmp/efa \
- && rm -rf /tmp/aws-efa-installer-${EFA_VERSION}.tar.gz
+ && rm -rf /tmp/aws-efa-installer-${EFA_VERSION}.tar.gz \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
```
Comment (jeet4320) [inline pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu:134]: can you add below change to this file?

https://github.com/aws/deep-learning-containers/blob/master/pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu#L199

**Review by jeet4320 — CHANGES_REQUESTED** [review]

File: pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+
+# Install OpenSSH for MPI to communicate between containers, allow OpenSSH to talk to containers without asking for confirmation
+RUN apt-get update \
+ && apt-get install -y  --allow-downgrades --allow-change-held-packages --no-install-recommends \
+ && apt-get install -y --no-install-recommends openssh-client openssh-server \
+ && mkdir -p /var/run/sshd \
+ && cat /etc/ssh/ssh_config | grep -v StrictHostKeyChecking > /etc/ssh/ssh_config.new \
+ && echo "    StrictHostKeyChecking no" >> /etc/ssh/ssh_config.new \
+ && mv /etc/ssh/ssh_config.new /etc/ssh/ssh_config \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (tejaschumbalkar) [inline pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu:406]: lets add apt-get clean

**Comment by tejaschumbalkar** [comment]
Lets enable SM test by changing `sagemaker_local_tests = true` [here](https://github.com/aws/deep-learning-containers/blob/master/dlc_developer_config.toml#L41) and `sagemaker_remote_tests = "standard"`  [here](https://github.com/aws/deep-learning-containers/blob/master/dlc_developer_config.toml#L48)

File: pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu (hunk 1 above)
Comment (ydaiming) [reply pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu:133]: Discussed offline, already addressed.

File: pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
```
[the remark this answers is not in the document]
Comment (ydaiming) [reply pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu]: addressed in latest commits

File: pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu (hunk 2 above)
Comment (ydaiming) [reply pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu]: addressed in latest commits

File: pytorch/inference/docker/1.11/py3/cu113/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+# |_____|____/  |___|_| |_| |_|\__,_|\__, |\___| |_| \_\___|\___|_| .__/ \___|
+#                                    |___/                        |_|
+##############################################################################
+
+FROM nvidia/cuda:11.3.1-base-ubuntu20.04 AS e3
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+ARG PYTHON
```
[the remark this answers is not in the document]
Comment (ydaiming) [reply pytorch/inference/docker/1.11/py3/cu113/Dockerfile.gpu:24]: It has to be re-declared in each stage

**Comment by tejaschumbalkar** [comment]
LGTM

**Review by jeet4320 — APPROVED** [review]

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**